File: G03 - 2025 TC - Informe - Enunciado Práctico TP 02 y TP 03/p1/Factorial.java
public class Factorial {
    private int numero;

    public Factorial(int numero) {
        this.numero = numero;
    }

    public int calcular() {
        int resultado = 1;
        for (int i = 2; i <= numero; i++) {
            resultado *= i;
        }
        return resultado;
    }

    public static void main(String[] args) {
        Factorial f = new Factorial(5);
        System.out.println(f.calcular()); // Salida: 120
    }
}

File: G03 - 2025 TC - Informe - Enunciado Práctico TP 02 y TP 03/p1/factorial.pl
factorial(0, 1).
factorial(N, F) :-
    N > 0,
    N1 is N - 1,
    factorial(N1, F1),
    F is N * F1.

:- initialization(main).

main :- 
    factorial(5, F),
    write('El factorial de 5 es: '), write(F), nl.